magento/graphql-ce
# Quote masked id creation logic moved to create cart resolver

### Description
This PR brings a few improvements to the empty cart creation process introduced in https://github.com/magento/graphql-ce/pull/124. 

The masked quote id creation is moved from `\Magento\Quote\Model\QuoteIdToMaskedQuoteId` to cart creation resolver in order to preserve CQRS principle.
Also, the masked quote id is recreated on every request for empty cart creation. In that way, we have duplicates in `quote_id_mask` table. 
An additional logic has been added that checks for existing id masks for a requested quote.

### Manual testing scenarios
1. Create empty cart for logged in customer using GraphQL request.
2. Check `quote_id_mask` database table. A new record should be created.
3. Make the same request once again.
4. No extra records should appear in the `quote_id_mask` table.
